spec/factories/accessories.rb

```diff
@@ -1,4 +1,14 @@
 FactoryGirl.define do
+  factory :instrument_with_accessory, parent: :setup_instrument do
+    transient do
+      accessory { create :setup_item, facility: facility }
+    end
+
+    after(:create) do |instrument, evaluator|
```

Comment:
Yeah. If I remember from Friday, what was happening is the `save` on instrument wasn't persisting the `product_accessory`.